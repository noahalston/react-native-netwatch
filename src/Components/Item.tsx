import * as React from 'react';
import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';
// @ts-ignore
import { Appbar, Searchbar, Surface, List, Divider } from 'react-native-paper';
import { Request } from '../Core/Request';
import { Status } from './Status';
import { getTime } from '../Utils/helpers';
import url from 'url';

export interface IProps {
  item: Request;
  onPress: () => void;
}

export const Item: React.FC<IProps> = ({ item, onPress }) => {
  let urlObject = url.parse(item.url);
  return (
    <TouchableOpacity onPress={() => onPress()} style={styles.listItemContainer}>
      <>
        <Status item={item}></Status>
        <View style={styles.url}>
          <View style={styles.domain}>
            <Text numberOfLines={1} style={styles.titleStyle}>{`${urlObject.host}`}</Text>
            <Text style={styles.path}>{getTime(item.endTime)}</Text>
          </View>
          <Text numberOfLines={1} style={styles.descriptionStyle}>
            {urlObject.path}
          </Text>
        </View>
      </>
    </TouchableOpacity>
  );
};

export default Item;

const styles = StyleSheet.create({
  listItemContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    borderBottomColor: '#bdbdbd',
    borderBottomWidth: 0.5,
    backgroundColor: '#757575',
  },
  url: {
    flex: 1,
    paddingRight: 8,
  },
  domain: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  path: {
    color: 'white',
  },
  titleStyle: {
    paddingLeft: 16,
    color: 'white',
    fontWeight: 'bold',
  },
  descriptionStyle: {
    paddingLeft: 16,
    color: 'white',
  },
  divider: {
    backgroundColor: 'gray',
  },
});
