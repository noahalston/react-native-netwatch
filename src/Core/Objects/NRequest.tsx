import { ILog, LogType } from '../../types';

export class NRequest implements ILog {
  _id: number = -1;
  type: LogType = 'NR';
  startTime: number = Date.now();
  readyState: number = 0;
  url: string = '';
  shortUrl?: string = '';
  method: string = '';
  status: number = -1;
  endTime: number = 0;
  timeout?: number;
  dataSent: string = '';
  requestHeaders?: any;
  responseHeaders?: any;
  responseContentType?: string;
  responseSize?: number;
  responseType?: string;
  responseURL?: string;
  response: string = '';

  constructor(attributes?: any) {
    Object.assign(this, attributes);
  }
}

export default NRequest;
