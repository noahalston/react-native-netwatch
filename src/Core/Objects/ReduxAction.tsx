import { ILog, LogType } from '../../types';

interface IAction {
  type: string;
  payload: any;
}

export class ReduxAction implements ILog {
  _id: number = -1;
  startTime: number = Date.now();
  type: LogType = 'REDUX';
  action: IAction = { type: '__ERROR:UNDEFINED__', payload: '' };

  constructor(attributes?: any) {
    Object.assign(this, attributes);
  }
}

export default ReduxAction;
