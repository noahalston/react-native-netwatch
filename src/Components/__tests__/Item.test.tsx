import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Item, IProps } from '../Item';
import { RNRequest } from '../../Core/Objects/RNRequest';

describe('Item test suite', () => {
  let component: ShallowWrapper;
  let props: IProps;
  const onPress = jest.fn();

  it('should render properly', () => {
    givenProps();
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  it('should render properly & press item', () => {
    givenProps();
    givenComponent();
    expect(component).toMatchSnapshot();

    whenPressingButton('itemTouchable-73');
    expect(onPress).toHaveBeenCalledTimes(1);
  });

  // GIVEN
  function givenComponent() {
    component = shallow(<Item {...props} />);
  }

  function givenProps() {
    props = {
      onPress,
      item: mockRequest,
    };
  }

  // UTILITIES

  function whenPressingButton(testId: string) {
    component.find(`[testID="${testId}"]`).simulate('press');
  }

  const mockRequest: RNRequest = {
    _id: 73,
    dataSent: 'dataSent',
    endTime: 1613477575757,
    method: 'GET',
    readyState: 4,
    response: 'response',
    responseContentType: 'application/json',
    responseSize: 0,
    responseType: 'blob',
    responseURL: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
    startTime: 1613477574742,
    status: 200,
    timeout: 0,
    type: 'RNR',
    url: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
  };
});
