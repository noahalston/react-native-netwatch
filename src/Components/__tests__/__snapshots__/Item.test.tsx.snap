// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item test suite should render properly & press item 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#757575",
      "borderBottomColor": "#bdbdbd",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
    }
  }
  testID="itemTouchable-73"
>
  <Status
    item={
      Object {
        "_id": 73,
        "dataSent": "dataSent",
        "endTime": 1613477575757,
        "method": "GET",
        "readyState": 4,
        "response": "response",
        "responseContentType": "application/json",
        "responseSize": 0,
        "responseType": "blob",
        "responseURL": "https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9",
        "startTime": 1613477574742,
        "status": 200,
        "timeout": 0,
        "type": "RNR",
        "url": "https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9",
      }
    }
  />
  <Component
    style={
      Object {
        "flex": 1,
        "paddingRight": 8,
      }
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Component
        numberOfLines={1}
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
            "paddingLeft": 16,
          }
        }
      >
        run.mocky.io
      </Component>
      <Component
        style={
          Object {
            "color": "white",
          }
        }
      >
        12:12:54
      </Component>
    </Component>
    <Component
      numberOfLines={1}
      style={
        Object {
          "color": "white",
          "paddingLeft": 16,
        }
      }
    >
      /v3/1a2d092a-42b2-4a89-a44f-267935dc13e9
    </Component>
  </Component>
</ForwardRef>
`;

exports[`Item test suite should render properly 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#757575",
      "borderBottomColor": "#bdbdbd",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
    }
  }
  testID="itemTouchable-73"
>
  <Status
    item={
      Object {
        "_id": 73,
        "dataSent": "dataSent",
        "endTime": 1613477575757,
        "method": "GET",
        "readyState": 4,
        "response": "response",
        "responseContentType": "application/json",
        "responseSize": 0,
        "responseType": "blob",
        "responseURL": "https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9",
        "startTime": 1613477574742,
        "status": 200,
        "timeout": 0,
        "type": "RNR",
        "url": "https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9",
      }
    }
  />
  <Component
    style={
      Object {
        "flex": 1,
        "paddingRight": 8,
      }
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Component
        numberOfLines={1}
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
            "paddingLeft": 16,
          }
        }
      >
        run.mocky.io
      </Component>
      <Component
        style={
          Object {
            "color": "white",
          }
        }
      >
        12:12:54
      </Component>
    </Component>
    <Component
      numberOfLines={1}
      style={
        Object {
          "color": "white",
          "paddingLeft": 16,
        }
      }
    >
      /v3/1a2d092a-42b2-4a89-a44f-267935dc13e9
    </Component>
  </Component>
</ForwardRef>
`;
